src/hotspot/share/gc/shenandoah/shenandoahFreeSet.hpp

```diff
@@ -205,18 +364,30 @@ class ShenandoahFreeSet : public CHeapObj<mtGC> {
   // ShenandoahOldCompactionReserve.  In a future planned enhancement, the reserve for old_collector set when the
   // evacuation reserves are unknown, is based in part on anticipated promotion as determined by analysis of live data
   // found during the previous GC pass which is one less than the current tenure age.
-  void rebuild(size_t young_cset_regions, size_t old_cset_regions, bool have_evacuation_reserves = false);
-
-  void move_collector_sets_to_mutator(size_t cset_regions);
-
-  void add_old_collector_free_region(ShenandoahHeapRegion* region);
+  void finish_rebuild(size_t young_cset_regions, size_t old_cset_regions, size_t num_old_regions,
```

Comment:
For the documentation comments, for the individual public methods themselves, viz. `prepare_to_rebuild`, `finish_rebuild`, `find_regions_with_alloc_capacity` etc., it's easier to have a terser comment that describes the parameters to each call and their side-effect/result.

Then, in the callers of these methods (the implementation of the rebuild) you can place the commentary about how the rebuild is happening by stitching these methods together. Otherwise the header file can be confusing in the form of an imperative narrative.

Reply:
I've taken a pass at improving descriptions of all arguments to these functions.